packages/core/types/ts3.6/effects.d.ts

```diff
@@ -583,9 +583,7 @@ export function cps<Ctx, Fn extends (this: Ctx, ...args: any[]) => void>(
   ...args: CpsFunctionParameters<Fn>
 ): CpsEffect<ReturnType<Fn>>
 
-export type CpsFunctionParameters<Fn extends (...args: any[]) => any> = Last<Parameters<Fn>> extends CpsCallback<any>
-  ? AllButLast<Parameters<Fn>>
-  : ['must provide a function with cps callback']
+export type CpsFunctionParameters<Fn extends (...args: any[]) => any> = AllButLast<Parameters<Fn>>
```

Comment:
Hmm, is this change correct? Now both 3.6 and 4.x **don't error** on the case that we are discussing and also we'd fail to error on this:
```ts
yield cps((a: number, b: string) => {}, 42)
```

Reply:
CI is passing for it: https://github.com/neurosnap/redux-saga/actions/runs/1917541095

Reply:
This is surprising - could you try to check why that is? Last time I've checked this didn't pass on the master branch but it did after your change, so perhaps this is a missing test case that we should add to guard us from regressions

Reply:
I can confirm that this example typechecks although it shouldn't:
[TS playground](https://www.typescriptlang.org/play?ts=3.6.3#code/C4TwDgpgBACgThSA7AJgHgCoFcwBtoQAewEqAzlAIZIgDaAugDRQCCKKpKAfFALxQAKAQH0AXK3admAOlnCAjOOx4IASj49qIdURLlBA2dLFQAlkgBmEOFABKEMllzB1vTTXUB+Ow6fAo4kgQAG7WANwAUBGgkFAAkmQAYuamJJg4+FC6nBRaDMzJSKkQzHGWKSQ8-ADeEVD1WQC2YKDihcWRDVBIAPZIAKLNrQZGJsr4ru7aWcQ5BibmVjbJcGTAMnKKZpbWPmuTVB5Q3gntafZrBRUl8eVFlQHdIeF1DYvX4mUW15EAvrTjAizfQMY5QADkECGIHBj0BMz0KAoAkWu36+EapBcoO86KhWNB2X08O84Pe9wgsPE4N6A2hVKeoTg9CiHAAxrhKAgoDFoMIADKUNbpFQIuZ5JhQAAiEAslD8fEZ1iqUFqXShLRA4hlcr8nQaZHMAHN8EoMkDERRaKibABlHr4PGYpDAehg+2OjFYx5BJn6+qNPymcRCUZmlQHLQ6YFI+biLQbYxbG1QAByukjR28guF6eIPG1svlzn922+FPDmSJsZROxsTqxqhxUAbLp9zzgfwB5rFILdpI1oAZ8OrVq0-YhhqQJspcJ7o8EKdb2Iny8JMYoJIh5OKDPBgecpgZvusLOi4GgOeAIqrG8OdHoKoFQuvgK4kXPsWE9iZZAgN4tcUaHyWAEG+Qhe1jCVFQYFU1QaQctVA2VTEIUtaUGTUQ0MORKzUDR72jS04zLXYVkuKBRmTOs0wzAiozBb8Oz-NA83WZDkHQeAUMIK5VmALgCyVTtXnqHcSHEeCunqfpbFsAB5WxqQAYWoXp-AQX9oGoMtrh5c1wVLLplPkqV+lTFgAFl+mpL5rkBQzRKgX4uxHO9aATSjZAlCdTmuADmBpPpMKHQLxMpITIXpM9eR8LSAMg3JgMfRUmPit8P1ijBKFMXA0H5RL71gxVQ1kXBxH5TNpgXIQAAt4xoRNgHEFMMCqrwoAwdsmSiWKWFwXAACEsGAK98sKiUVR-awWOy3K0Gm1Z-35QSuCicwSDgOU2WgZSwDIVSBoAI0oNkAGsFp4eCBGsOAejgBqQGYBBHGccRbFUcRgh6UwUFeNlqB23BPAET6oG+36Il+XqLygPakiwJA2WAUw+hgLlKExTayDQRIkEKnDpC5I0yEehgqpVfqhpGsb0bgTGIGx3GkEE1kIA5LloAsRHkdR-G2X25mCaMYnSaK+gKYEV4LCQNokEYV4RbgEnxHhxIeZRtGMaxmbmbWsGIb+iIBbIIRKECLBGiO6xmCO8Q1jgY0DmqX5mAAFgAJlUIA)